src/other-reprs.md

```diff
@@ -56,24 +56,27 @@ compiled as normal.)
 
 ## repr(transparent)
 
-This can only be used on structs with a single non-zero-sized field (there may
-be additional zero-sized fields). The effect is that the layout and ABI of the
-whole struct is guaranteed to be the same as that one field.
+This can only be used on structs with a single non-zero-sized field
+or enums with only one non-zero-sized field (there may be additional zero-sized fields).
```

Comment:
I think it might be good to be explicit that the enum can only have a single variant.  I might also rework this a little to avoid the duplication, so maybe something like:

```suggestion
`#[repr(transparent)]` can only be used on a struct or single-variant enum that has a single non-zero-sized field (there may be additional zero-sized fields).
```